### ISSUE TITLE
关于backward的问题

### ISSUE THREAD

**Issue by robotcator** [walkerning/nics_fix_pytorch#3]
你好，请教一下，这边有看到对forward的处理主要是get_fix_forward函数里面，没有看到关于backward的相关的特殊处理，请问定点训练里面，这部分是怎么做的呢？有相关的文章吗？

**Comment by walkerning** [comment]
定点训练的backward主要是straight through，也就是前向的round操作，在反向是直接把gradient，反向相当于一个identity 操作。搜straight through应该能搜到。

**Comment by robotcator** [comment]
了解，找到对应的相关代码了。还有一个问题，这个跟谷歌论文[1, 2]里面提到quantization awareness training的区别是什么呢？

[1] https://arxiv.org/abs/1712.05877
[2] https://www.tensorflow.org/lite/performance/model_optimization

**Comment by walkerning** [comment]
关键的区别是quantization scheme不同，它文中描述的每个weight/activation tensor有一个floating point的scale和一个zero-point，我们没有这个。讲道理这样肯定比我们的quantization scheme更powerful，不过这个和针对的硬件有关。

并且它文中第三节提到的两个implementation细节， 都是我们之前觉得有必要的，但是这个公开的代码里并没有支持。@ILTShade 孙博考虑加一下？
1. batchnorm folding，inference的时候batchnorm是和conv层fuse到一起的，所以为了定点仿真训练和实际定点inference的discrepancy尽量小，训练的时候对这两个层的定点应该一起做。
2. feature(i.e. activation)的quantization range每个图片不一样，确实应该要做一个类似moving average的方法来更新。

**Comment by robotcator** [comment]
感谢解答～
